www/js/main.js

```diff
@@ -100,46 +100,60 @@ angular.module('emission.main', ['emission.main.recent', 'emission.main.diary',
     $scope.emailLog = ControlHelper.emailLog;
     $scope.dark_theme = $rootScope.dark_theme;
 
-    
-    $scope.toggleLowAccuracy = function() {
+
+    $scope.getLowAccuracy = function() {
         //  return true: toggle on; return false: toggle off.
         if ($scope.settings.collect.config == null) {
             return false; // config not loaded when loading ui, set default as false
         } else {
+            var accuracy = $scope.settings.collect.config.accuracy; 
+            var v;
+            for (var k in $scope.settings.collect.accuracyOptions) {
+                if ($scope.settings.collect.accuracyOptions[k] == accuracy) {
+                    v = k;
+                    break;
+                } 
+            }
             if ($scope.isIOS()) {
-                var accuracy = $scope.settings.collect.config.accuracy; 
-                return accuracy >= 100;
+                return v != "kCLLocationAccuracyBestForNavigation" && v != "kCLLocationAccuracyBest";
             } else if ($scope.isAndroid()) {
-                return accuracy != "PRIORITY_HIGH_ACCURACY";
+                return v != "PRIORITY_HIGH_ACCURACY";
             }
 
         }
     }
-    $scope.willUseLowAccuracy = function() {
+    $scope.setLowAccuracy = function() {
         var accuracy = $scope.settings.collect.config.accuracy;
         $scope.settings.collect.new_config = JSON.parse(JSON.stringify($scope.settings.collect.config));
+        var v;
+        for (var k in $scope.settings.collect.accuracyOptions) {
```

Comment:
You could also consider just prepopulating the string value when the config is read (in `getSettings`)